hazelcast/src/main/java/com/hazelcast/map/impl/wan/WanMapRemoveEvent.java

```diff
@@ -49,6 +56,12 @@ public Data getKey() {
         return key;
     }
 
+    @Nullable
+    @Override
+    public Object getEventObject() {
+        return serializationService.toObject(key);
```

Comment:
can we optimize this part? Now we deserialize key for every call.

Reply:
You're right. If you're ok with it, I might optimise it in a follow-up PR since I was already planning on addressing the issue of creating new entry instances for update events and the rabbit hole I got dragged into on slack.

Reply:
what about using `getEventObject` in also our implementation?

Reply:
We might, but only after I complete the follow-up change where we are able to expose `WanMapEntryView` to users. Also, that class needs to be able to both lazily deserialise and to provide internal methods to get the `Data` to avoid deserialization.

Reply:
should i expect the optimization in this PR?

Reply:
Yes, I'll add it.